### PR TITLE
Demonstrate Hierarchy Functions

### PR DESCRIPTION
**Pull request by etimr**
Provides an example of an hierarchy function as `ListReport` and `ValueHelp`

- currently only `READ`
- to [enable tree tables](https://ui5.sap.com//#/topic/7cf7a31fd1ee490ab816ecd941bd2f1f) as `ListReport` add to `manifest.json`:
```
"tableSettings": {
     "type": "TreeTable",
     "hierarchyQualifier": "NodesHierarchy",
   ...
}
```
- to enable tree table for ValueHelp hierarchical entity should be annotated with `PresentationVariant` and root entity with `PresentationVariantQualifier`

### CODE REVIEW

File: db/books.cds [hunk 1]
```diff
@@ -46,9 +45,11 @@ annotate Authors with
...
-    key ID       : Integer;
-        parent   : Association to Genres;
-        children : Composition of many Genres
-                       on children.parent = $self;
+entity Genres {
+    key ID      : Integer;
+    name        : localized String(255);
+    descr       : localized String(1000);
+    parent    : Association to Genres;
+    children : Composition of many Genres
```
Comment (agoerler) [inline db/books.cds:53]:
```suggestion
    descr       : localized String(1000);
    parent     : Association to Genres;
    children   : Composition of many Genres
```

File: srv/admin-service.cds [hunk 2]
```diff
@@ -1,34 +1,43 @@
 using {sap.common.Languages as CommonLanguages} from '@sap/cds/common';
 using {my.bookshop as my} from '../db/index';
 using {sap.changelog as changelog} from 'com.sap.cds/change-tracking';
-using {sap.attachments.Attachments} from `com.sap.cds/cds-feature-attachments`;
+using {my.common.Hierarchy as Hierarchy} from './hierarchy';
+using {sap.attachments.Attachments} from`com.sap.cds/cds-feature-attachments`;
```
Comment (agoerler) [inline srv/admin-service.cds:5]:
```suggestion
using {sap.attachments.Attachments} from `com.sap.cds/cds-feature-attachments`;
```

File: srv/review-service.cds [hunk 3]
```diff
@@ -13,6 +13,9 @@ service ReviewService {
     @readonly
     entity Authors as projection on my.Authors;
 
+    @readonly
+    entity Genres     as projection on my.Genres;
```
Comment (agoerler) [inline srv/review-service.cds:17]:
```suggestion
    entity Genres  as projection on my.Genres;
```

File: srv/src/main/java/my/bookshop/handlers/HierarchyHandler.java [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
+
+import cds.gen.adminservice.AdminService_;
+import cds.gen.adminservice.GenreHierarchy;
+import cds.gen.adminservice.GenreHierarchy_;
+
+import static cds.gen.adminservice.AdminService_.GENRE_HIERARCHY;
+
```
Comment (agoerler) [inline srv/src/main/java/my/bookshop/handlers/HierarchyHandler.java:43]: please add javadoc which explains that this handler is only a stand-in as long as we can't handler hierarchy transformations generically on non-HANA (H2)

please also add a warning that this handler is not necessarily functionally correct and should not be used as a blue-print

File: srv/src/main/resources/application.yaml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 ---
 logging:
   level:
-    '[com.sap.cds.auditlog]': DEBUG
+    com.sap.cds.auditlog: DEBUG
+    com.sap.cds.persistence.sql: DEBUG
```
Comment (agoerler) [inline srv/src/main/resources/application.yaml]: undo ?

File: srv/admin-service.cds [hunk 6]
```diff
@@ -1,34 +1,43 @@
...
 
 extend my.Orders with changelog.changeTracked;
 
-@path : 'admin'
-service AdminService @(requires : 'admin') {
-  entity Books   as projection on my.Books excluding { reviews } actions {
-    action addToOrder(order_ID : UUID, quantity : Integer) returns Orders;
-  }
+@path: 'admin'
```
Comment (agoerler) [inline srv/admin-service.cds:9]: there are many unrelated formatting changes in this file. Please minimized the diff with `main`

File: pom.xml [hunk 7]
```diff
@@ -23,7 +23,7 @@
 
 		<!-- DEPENDENCIES VERSION -->
 		<jdk.version>21</jdk.version>
-		<cds.services.version>3.5.0</cds.services.version>
+		<cds.services.version>3.6.0-m2451</cds.services.version>
```
Comment (agoerler) [inline pom.xml:26]: we can merge to `main` only once 3.6.0 is released

**Review by agoerler — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 7 above)
Comment (eugene-andreev) [inline pom.xml:26]:
```suggestion
		<cds.services.version>3.6.0</cds.services.version>
```

File: db/books.cds (hunk 1 above)
Comment (agoerler) [reply db/books.cds:53]: formatted in VS code

File: srv/review-service.cds (hunk 3 above)
Comment (agoerler) [reply srv/review-service.cds:17]: formatted

File: srv/admin-service.cds (hunk 2 above)
Comment (agoerler) [reply srv/admin-service.cds:5]: formatted

File: db/books.cds [hunk 8]
```diff
@@ -37,17 +36,21 @@ entity Authors : cuid, managed {
...
+    EntitySemantics: 'DataSubject'
+} {
+    ID @PersonalData.FieldSemantics: 'DataSubjectID';
+    name @PersonalData.IsPotentiallySensitive;
 }
 
 /**
  * Hierarchically organized Code List for Genres
  */
-entity Genres : sap.common.CodeList {
```
Comment (beckermarc) [inline db/books.cds:49]: Why can't you use CodeList from sap.common?

File: README.md [hunk 9]
```diff
@@ -138,6 +141,14 @@ are defined for local development:
 - User: `user`, password: `user` to browse books
 - User: `admin`, password: `admin` to manage books and orders
 
+### Testing in hybrid mode
+
+You can test the `GenreHierarchyTest` on H2 using the profile `default` as well as on HANA using the profile `hybrid`
```
Comment (beckermarc) [inline README.md:146]: I'd rather recommend usage of existing profiles in application.yaml, e.g. `cloud`, which expects HANA.

File: db/books.cds (hunk 8 above)
Comment (agoerler) [reply db/books.cds:49]: @OlenaTi - do you recall why we changed this?

Comment (agoerler) [reply db/books.cds:49]: -> https://github.com/SAP-samples/cloud-cap-samples-java/pull/419/files#diff-d92b8eb29957a03ad9d23b01feff79da293e30183d0b8eac5d893d95632e30dfR51

Comment (etimr) [reply db/books.cds:49]: https://github.com/SAP-samples/cloud-cap-samples-java/pull/388#issue-2624645140
